src/Http/Http.Api.Abstractions/src/IFromRouteAttribute.cs

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Http.Api.Abstractions
+{
+    /// <summary>
+    /// Interface marking attributes that specify a parameter or property should be bound using route-data from the current request.
```

Comment:
> specify a parameter or property


I think we would only want to support this feature for parameters. This is a bit problematic because the actual attributes allow being applied to properties. Would we write an analyzer to detect it?

Reply:
Listing other sources we would want to support:

* Service
* Query
* Header
* File 

What this leaves out: [FromForm] / [ModelBinder] / [Bind] etc.

Reply:
The doc comment is just a bit of over-copy-pasting. Obviously, it can only work for parameters with MapAction. An analyzer to detect trying to use `[FromRoute]` on a property outside of a controller class probably would be helpful.

I think we want to support `[FromForm]` for non-complex types. https://github.com/dotnet/aspnetcore/issues/29684#issuecomment-769769659

Reply:
My primary push-back to form would be that it's not very common to have a form with simple data types. Given that there's no JSON, it seems like a fairly narrow scenario to try and be bothered with.

Reply:
Files?

Reply:
Files are in my list. We should be able to support both IFormFileCollection (bind everything) and IFormFile. If we get super lucky we could figure out how to shoe horn Multipart content containing JSON and files.